src/cpp/session/modules/SessionPackages.R

```diff
@@ -1429,7 +1429,16 @@ if (identical(as.character(Sys.info()["sysname"]), "Darwin") &&
    # is the directory old? if so, assume the prior R process launched
    # to produce available.packages crashed or similar
    info <- file.info(dir)
-   diff <- difftime(Sys.time(), info$mtime, units = "secs")
+   time <- info$mtime
+   
+   # in some cases, mtime may not be available -- in that case, fall
+   # back to the time that was serialized when the process was launched
+   # https://github.com/rstudio/rstudio/issues/4312
+   if (is.na(time))
+      time <- readRDS(file.path(dir, "time.rds"))
```

Comment:
Is there a pretty strong guarantee that the file exists and is readable at this point? (if not we might need a last-resort that just makes a safe assumption)

Reply:
I can't imagine how the directory could exist without the associated `time.rds` file also existing (since they're created at the same time) but better safe than sorry -- I've added one final fallback that at least ensures that the user doesn't get polluted with warnings.